public/app/features/explore/Logs/LogsTableWrap.tsx

```diff
@@ -44,14 +45,19 @@ type fieldNameMetaStore = Record<fieldName, fieldNameMeta>;
 
 export function LogsTableWrap(props: Props) {
   const { logsFrames } = props;
+
   // Save the normalized cardinality of each label
   const [columnsWithMeta, setColumnsWithMeta] = useState<fieldNameMetaStore | undefined>(undefined);
 
   // Filtered copy of columnsWithMeta that only includes matching results
   const [filteredColumnsWithMeta, setFilteredColumnsWithMeta] = useState<fieldNameMetaStore | undefined>(undefined);
 
   const height = getTableHeight();
-  const dataFrame = logsFrames[0];
+  // The refId of the current frame being displayed
+  const [currentFrameRef, setCurrentFrameRef] = useState<string | undefined>(logsFrames[0].refId);
+  const [dataFrame, setDataFrame] = useState(
+    logsFrames.find((frame) => frame.refId === currentFrameRef) ?? logsFrames[0]
```

Comment:
I am wondering if we need 2 states for this. It looks like whenever 1 change, the other needs to change as well. Wouldn't it be simpler if we just had something like `currentDataFrame` and just used its `refId` whenever needed?

Reply:
That would be simpler in the code, however that introduces a flicker when changing dataFrames. I get this same bug when removing either of these, I think it's because when we update the ref, that updates the dataFrame (or the frame that updates the ref), is causing additional re-renders before the new dataframe is applied to the state.

https://github.com/grafana/grafana/assets/109082771/8f990df8-8eeb-44d6-a941-67a8acdfa331

Reply:
Yeah I can see that. 😞 Then maybe just adding a comment on why do we need both.

Reply:
Added!